trillian/migrillian/election/etcd/election.go

```diff
@@ -105,28 +76,19 @@ func (e *Election) Await(ctx context.Context) (context.Context, error) {
 		}
 	}()
 
-	e.cancel, e.done = cancel, done
 	return cctx, nil
 }
 
-// Resign releases mastership for this instance and cancels the mastership
-// context. The instance can be elected again using Await.
+// Resign releases mastership for this instance. The instance can be elected
+// again using Await. Idempotent, might be useful to retry if fails.
 func (e *Election) Resign(ctx context.Context) error {
-	err := e.election.Resign(ctx)
-	// Cancel after Resign to tolerate ctx being the mastership context.
-	if e.cancel != nil {
-		e.cancel()
-		// Ignore cxt.Done() while waiting for e.done, to tolerate ctx being the
-		// canceled mastership context. This wait will reliably terminate anyway.
-		<-e.done
-	}
-	return err
+	return e.election.Resign(ctx)
 }
 
-// Close cancels the mastership context, permanently stops participating in
-// election, and releases the resources. It does best effort on resigning
-// despite potential cancelation of the passed in context. No other method
-// should be called after Close.
+// Close permanently stops participating in election, and releases the
+// resources. It does best effort on resigning despite potential cancelation of
+// the passed in context, so that other instances can overtake mastership
+// faster. No other method should be called after Close.
```

Comment:
(nit) Copy-pasting the text from the `interface` is likely to lead to the copies getting out-of-sync; ideally the docs for the `interface` would describe the behaviour of the interface from a user's perspective, and the text in the individual implementations can be shorter and just concentrate on the implementation-specific aspects.

Reply:
Good point. Done.